packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -134,12 +146,19 @@ function SelectionTool(props: Props) {
 
   // Compute effective selection
   const selection = useMemo(
-    () => rawSelection && transformSelectionRef.current(rawSelection),
-    [rawSelection, transformSelectionRef]
+    () => rawSelection && transformRef.current(rawSelection),
+    [rawSelection, transformRef]
   );
 
-  // Keep track of previous effective selection state
+  // Determine if effective selection respects the minimum size threshold
+  const isValid = useMemo(
+    () => !!selection && validateRef.current(selection),
+    [selection, validateRef]
+  );
+
+  // Keep track of previous effective selection and validity
   const prevSelection = usePrevious(selection);
+  const prevIsValid = usePrevious(isValid);
```

Comment:
Very slightly annoying to have to keep track of the previous validation state as well (for `onSelectionEnd` and `onValidSelection`), but the alternatives are awful:

- like adding `isValid` to the `Selection` mode, which doesn't make sense since `transform()` receives a `Selection` object but the validity of the selection is checked afterwards;
- or attempting yet again to store the transformed selection and validity flag in local state...

Reply:
:shrug: